tests/sklearn/test_sklearn_models.py

```diff
@@ -2352,3 +2354,73 @@ def test_xgb_serialization_errors(model_class, param, error_message):
         with pytest.raises(NotImplementedError, match=error_message):
             model = instantiate_model_generic(model_class, 5, **param)
             model.dumps()
+
+
+@pytest.mark.parametrize(
+    "model_class, parameters", get_sklearn_tree_models_and_datasets(True, True)
+)
+@pytest.mark.parametrize("n_bits", [4, 8, 12])
+def test_tfhers_inputs_outputs_trees(model_class, parameters, n_bits, load_data):
+    """Check that 8b tree-based classifiers work with TFHE-rs inputs/outputs."""
+
+    x, y = get_dataset(model_class, parameters, n_bits, load_data, True)
+
+    # Use a single example in FHE to make the test fast enough for the CI
+    fhe_test_data = x[0:1, :]
+
+    model = instantiate_model_generic(model_class, n_bits=n_bits)
+    # Fit the model to create the equivalent sklearn model
+    model.fit(x, y)
+
+    # If the model is not supported or if the n_bits is not supported
+    # an error is raised
+    if not n_bits == 8 or is_regressor_or_partial_regressor(model_class):
+        with pytest.raises(AssertionError, match=".*supported for 8-bit tree-based.*"):
+            model.compile(x, ciphertext_format=CiphertextFormat.TFHE_RS)
+        return
+
+    # Check that we can first compile to Concrete, then to
+    # TFHE-rs input/outputs then to concrete again
+    model.compile(x)
```

Comment:
You could add model.predict with concrete maybe and add this to the comparisons. We expect them to be equal to disable and to execute with tfhe-rs input

Reply:
ok, as Benoit asks for it too I'll add it